### PR TITLE
MGMT-7084: Validate clusterDeployment status becomes installed

### PR DESCRIPTION
**Pull request by osherdp**
# Description

Make sure hive have acknowledged installation and marked cluster deployment as installed.

# What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [ ] None

# How was this code tested?

Please, select one or more if needed:

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

# Assignees

Please, add one or two reviewers that could help review this PR. Use `/assign` if you want to assign
this PR directly to someone.

/cc @YuviGold 
/cc @filanov 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: deploy/operator/ztp/deploy_spoke_cluster.sh [hunk 1]
```diff
@@ -61,3 +61,13 @@ echo "All ${CONTROL_PLANE_COUNT} agents have joined!"
 
 wait_for_condition "agentclusterinstall/${ASSISTED_AGENT_CLUSTER_INSTALL_NAME}" "Completed" "60m" "${ASSISTED_NAMESPACE}"
 echo "Cluster has been installed successfully!"
+
+installed=$(oc get -n ${ASSISTED_NAMESPACE} clusterdeployment/${ASSISTED_CLUSTER_DEPLOYMENT_NAME} \
```
Comment (YuviGold) [inline deploy/operator/ztp/deploy_spoke_cluster.sh:65]: Could you use `wait_for_condition`?

Comment (osherdp) [reply deploy/operator/ztp/deploy_spoke_cluster.sh:65]: no, first thing I checked
it's under spec, not on conditions

Comment (rollandf) [reply deploy/operator/ztp/deploy_spoke_cluster.sh:65]: This is not a condition so we cannot use `wait_for_condition`, but we should have some retries and not fail immediately.

Comment (filanov) [reply deploy/operator/ztp/deploy_spoke_cluster.sh:65]: +1 for retry

Comment (osherdp) [reply deploy/operator/ztp/deploy_spoke_cluster.sh]: done

**Pull request merged**